pySOT/strategy.py

```diff
@@ -850,3 +850,447 @@ def generate_evals(self, num_pts):
         else:
             for i in range(num_pts):
                 self.batch_queue.append(np.copy(np.ravel(new_points[i, :])))
+
+
+POSITIVE_INFINITY = float("inf")
+import scipy.spatial as scp
+from pySOT.utils import nd_sorting, taboo_region, dynamic_taboo_region
+from pySOT.auxiliary_problems import candidate_sop
+
+class SopRecord():
```

Comment:
Agreed on making class private. It is not a POAP record because it has additional unique attributes, and only shares x and f(x) values with POAP record. I have made some attributes private and implemented getter and update functions for some attributes